### PR TITLE
Parallelize S3 pull/push files + various bug fixes.

### PR DESCRIPTION
**Pull request by didip**
This is PR is essentially a merge of changes between 2 level of forks (aldrinleal and New Relic).

The changelog of what we, New Relic, made:
- Use `go fmt` on all files.
- Use the official `fsouza/go-dockerclient`
- Set `GOMAXPROCS` to consume all available CPU.
- Pulling files from S3 and pushing to S3 are parallelized using fixed pool of goroutines, currently hard-coded at 50.
- Removed the ability to set S3 prefix path. It makes debugging image name difficult.
- Logging functionality is now standarized to using `fmt` library.
- Added various code cleanup and test fixes.

### CODE REVIEW

File: remote/s3_test.go [hunk 1]
```diff
@@ -9,16 +9,16 @@ import (
...
-	"github.com/lachie/goamz/s3"
-	"github.com/lachie/goamz/testutil"
-	. "launchpad.net/gocheck"
+	"github.com/crowdmob/goamz/aws"
+	"github.com/crowdmob/goamz/s3"
+	"github.com/crowdmob/goamz/testutil"
+	. "gopkg.in/check.v1"
 
-  "github.com/blake-education/dogestry/config"
+	"github.com/newrelic-forks/dogestry/config"
```
Comment (relistan) [inline remote/s3_test.go:21]: Doubt this can go upstream like this.

File: Dockerfile [hunk 2]
```diff
@@ -1,19 +1,8 @@
...
-      ca-certificates \
-      --no-install-recommends
+MAINTAINER Aldrin Leal <aldrin@leal.eng.br>
 
-RUN curl -s https://go.googlecode.com/files/go1.2.linux-amd64.tar.gz | tar -v -C /usr/local -xz
-ENV	PATH	/usr/local/go/bin:$PATH
-ENV	GOPATH	/go:/go/src/github.com/blake-education/dogestry/vendor/go
-ADD . /go/src/github.com/blake-education/dogestry
+RUN go get github.com/fsouza/go-dockerclient
+RUN go get -v github.com/newrelic-forks/dogestry/dogestry
```
Comment (relistan) [inline Dockerfile:6]: Again...

File: cli/cli.go [hunk 3]
```diff
@@ -1,8 +1,8 @@
 package cli
 
 import (
-	"github.com/blake-education/dogestry/config"
 	docker "github.com/fsouza/go-dockerclient"
+	"github.com/newrelic-forks/dogestry/config"
```
Comment (relistan) [inline cli/cli.go:5]: Here, too.

File: cli/pull.go [hunk 4]
```diff
@@ -8,8 +8,8 @@ import (
 	"os/exec"
 	"path/filepath"
 
-	"github.com/blake-education/dogestry/remote"
 	docker "github.com/fsouza/go-dockerclient"
+	"github.com/newrelic-forks/dogestry/remote"
```
Comment (relistan) [inline cli/pull.go:12]: ...

**Comment by relistan** [comment]
There are a number of references in the PR to our fork. This is kind of annoying because we can't run them with `blake-education` instead of `newrelic` until this is merged.  Maybe these can be fixed at merge time?

**Comment by didip** [comment]
Closed and replaced with PR https://github.com/blake-education/dogestry/pull/11

**Pull request closed without merging**